### PR TITLE
schema changes

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/ploneorg/releasesecurityinfo/__init__.py [hunk 1]
```diff
@@ -22,3 +22,39 @@ def state_vocabulary(context):
...
+def ImpactVocabulary(context):
+    return SimpleVocabulary([
+        SimpleTerm("N", "low", title="None"),
+        SimpleTerm("P", "medium", title="Partial"),
+        SimpleTerm("C", "high", title="Complete"),
+    ])
+
+
+@provider(IVocabularyFactory)
+def ComplexityVocabulary(context)
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/__init__.py]: syntax error, needs to end with :

File: src/ploneorg/releasesecurityinfo/__init__.py [hunk 2]
```diff
@@ -22,3 +22,39 @@ def state_vocabulary(context):
...
+    return SimpleVocabulary([
+        SimpleTerm("H", "low", "High"),
+        SimpleTerm("M", "medium", "Medium"),
+        SimpleTerm("L", "high", "Low")
+    ])
+
+
+@provider(IVocabularyFactory)
+def AccessVectorVocabulary(context)
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/__init__.py]: same here, needs a :

File: src/ploneorg/releasesecurityinfo/__init__.py [hunk 3]
```diff
@@ -22,3 +22,39 @@ def state_vocabulary(context):
...
+    return SimpleVocabulary([
+        SimpleTerm("L", "low", "Local"),
+        SimpleTerm("A", "medium", "Adjacent Network"),
+        SimpleTerm("N", "high", "Network")
+    ])
+    
+
+@provider(IVocabularyFactory)
+def AuthenticationVocabulary(context)
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/__init__.py]: and also here.

File: src/ploneorg/releasesecurityinfo/__init__.py [hunk 4]
```diff
@@ -22,3 +22,39 @@ def state_vocabulary(context):
         SimpleTerm(value=u'OBSOLETE', title=u'Obsolete'),
         SimpleTerm(value=u'FUTURE', title=u'Future'),
     ])
+
+
+@provider(IVocabularyFactory)
+def ImpactVocabulary(context):
+    return SimpleVocabulary([
+        SimpleTerm("N", "low", title="None"),
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/__init__.py]: please use one quote style for all strings, Python recommends ' not "

File: src/ploneorg/releasesecurityinfo/contents.py [hunk 5]
```diff
@@ -31,13 +33,93 @@ class Release(Item):
...
+        # Hotfixes have their ID generated from their release date.
+        return self.release_date.strftime("%Y%m%d")
+
+    def released(self):
+        workflowTool = getToolByName(self, "portal_workflow")
+        status = workflowTool.getStatusOf("hotfix_workflow", self)
+        state = status["review_state"]
+        return state
+
+    def setTitle(self, title):
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/contents.py:73]: decorator is missing to take effect

File: src/ploneorg/releasesecurityinfo/field.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+from hashlib import sha1, md5
+from zope.interface import implements
+from zope import schema
+from zope.schema.fieldproperty import FieldProperty
+from plone.namedfile.interfaces import INamedFile, INamedFileField
+from plone.namedfile.field import NamedFile
+from plone.namedfile.file import NamedFile as FileValueType
+
+
+class IChecksummedFile(INamedFile):
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/field.py]: didn't we spoke that no file field is needed? why did you add those File field elements?

File: src/ploneorg/releasesecurityinfo/interfaces.py [hunk 7]
```diff
@@ -13,33 +17,44 @@ class IReleaseFolder(model.Schema):
     """
     Schema for ReleaseFolder.
     """
+    model.load("models/releasefolder.xml")
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/interfaces.py]: this is redundant with the settings in profiles.

**Review by loechel — CHANGES_REQUESTED** [review]
sorry this pull request is one of the worst I have to review from you. Please look at all comments and fix them.

I do expect you to run all tests localy and at least start up an instance and check if it starts. This Pull Request has so many errors that it could never pass any trivial check.

Please fix all problems and flake8&isort errors, and then inform me again.

File: src/ploneorg/releasesecurityinfo/__init__.py [hunk 8]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/__init__.py:1]: why is all of this in __init__.py an not in a separate vocabularies.py file?

File: src/ploneorg/releasesecurityinfo/configure.zcml [hunk 9]
```diff
@@ -38,4 +38,31 @@
...
+      component="ploneorg.releasesecurityinfo.AccessVectorVocabulary" />
+
+  <utility
+      name="ploneorg.releasesecurityinfo.AuthenticationVocabulary"
+      provides="zope.schema.interfaces.IVocabularyFactory"
+      component="ploneorg.releasesecurityinfo.AuthenticationVocabulary" />
+
+  <utility
+        provides="zope.schema.interfaces.IVocabularyFactory"
+        component="ploneorg.releasesecurityinfo.versions.plone_version_vocabulary"
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/configure.zcml]: all vocabularies come from one source except one, why?

File: src/ploneorg/releasesecurityinfo/versions.py [hunk 10]
```diff
@@ -0,0 +1,12 @@
+from plone.registry.interfaces import IRegistry
```
Comment (loechel) [inline src/ploneorg/releasesecurityinfo/versions.py]: why is this one special file?

**Review by loechel — CHANGES_REQUESTED** [review]
I do have a few question on style.
I know that you copy that from the original sources, but did not adapt the things learned from ploneorg.addonlisting about clean code and structures.

**Comment by ghost** [comment]
issue:
following error raised in releasefolder browserview:
**Refused to display 'https://launchpad.net/plone/+series' in a frame because it set 'X-Frame-Options' to 'SAMEORIGIN'**

**Comment by loechel** [comment]
Hi Pavi,
the X-Frame-Options are logical, as the browsers deny all cross domain requests for security reasons. but no worry that could be handled, we could either load and save local the element or do a  proxy request on the server side.

**Comment by loechel** [comment]
there has been some errors on this views, as missing schema fields that get referenced block the instance.

For now I would say this pull is ok, as it was focused on the schema. for the browser views that makes more that just printing the fields, make a new pull request please.

**Review by loechel — APPROVED** [review]
